### PR TITLE
Additional tracking for Google Sign In.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Issue: https://github.com/wordpress-mobile/WordPress-iOS/issues/14592
WPiOS PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/14611

Additional tracking for Google Sign In, and some enhancements to make the new analytics tracker easier to use in client Apps.

## Changes:

1. Renames `AnalyticsTracker` to `AuthenticatorAnalyticsTracker`.  The reason this is done is because the `WordPressAuthenticator` module name cannot be used to prefix the module classes.  This is because the module contains a class named `WordPressAuthenticator` (same name as the module) and this causes issues in client Apps when trying to use the module name as a class prefix.
2. Makes some enums in `AuthenticatorAnalyticsTracker.swift` public so that their usable from client Apps.
3. Sets the `.default` analytics tracking source when appropriate.
4. Adds a lot of tracking code.
5. Temporarily guards the unified-tracking so that we only track if `enableUnifiedAuth` is true.

## Testing:

Please see the testing steps in the WPiOS PR.

### CODE REVIEW

**Comment by diegoreymendez** [comment]
Hey @mindgraffiti , @ScoutHarris.  I'm temporarily removing this from review to add more successful test cases for the remaining Sign In flows.

File: WordPressAuthenticator/Services/LoginFacade.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
+//
+//  LoginFacade.swift
+//  WordPressAuthenticator
+//
+//  Created by Diego Rey Mendez on 12/08/2020.
+//  Copyright © 2020 Automattic. All rights reserved.
```
Comment (ScoutHarris) [inline WordPressAuthenticator/Services/LoginFacade.swift:6]: Can you remove this? We don't normally leave them.

File: WordPressAuthenticator/Signin/LoginEmailViewController.swift [hunk 2]
```diff
@@ -213,12 +213,16 @@ open class LoginEmailViewController: LoginViewController, NUXKeyboardResponder {
             }
 
             vc.googleTapped = { [weak self] in
+                guard let self = self else { return }
```
Comment (ScoutHarris) [inline WordPressAuthenticator/Signin/LoginEmailViewController.swift:216]: Per our style guide, `return` should be on a separate line. 
```
guard let self = self else { 
return 
}
```

File: WordPressAuthenticator/Signin/LoginPrologueViewController.swift [hunk 3]
```diff
@@ -138,12 +143,14 @@ class LoginPrologueViewController: LoginViewController {
         }
 
         vc.googleTapped = { [weak self] in
+            guard let self = self else { return }
```
Comment (ScoutHarris) [inline WordPressAuthenticator/Signin/LoginPrologueViewController.swift:146]: Same here - `return` should be on a separate line.

File: WordPressAuthenticator/Unified Auth/View Related/2FA/TwoFAViewController.swift [hunk 4]
```diff
@@ -387,7 +388,10 @@ private extension TwoFAViewController {
         cell.configureButton(text: WordPressAuthenticator.shared.displayStrings.textCodeButtonTitle)
 
         cell.actionHandler = { [weak self] in
-            self?.requestCode()
+            guard let self = self else { return }
```
Comment (ScoutHarris) [inline WordPressAuthenticator/Unified Auth/View Related/2FA/TwoFAViewController.swift:391]: Same here - `return` should be on a separate line.

File: WordPressAuthenticator/Unified Auth/View Related/Password/PasswordViewController.swift [hunk 5]
```diff
@@ -103,6 +103,13 @@ class PasswordViewController: LoginViewController {
     }
 
     override func displayError(message: String, moveVoiceOverFocus: Bool = false) {
+        // The reason why this check is necessary is that we're calling this method
+        // with an empty error message when setting up the VC.  We don't want to track
+        // an empty error when that happens.
+        if message.count > 0 {
```
Comment (ScoutHarris) [inline WordPressAuthenticator/Unified Auth/View Related/Password/PasswordViewController.swift:109]: Just a suggestion - `isEmpty` is more efficient as it doesn't have to count.

File: WordPressAuthenticator/Services/LoginFacade.swift (hunk 1 above)
Comment (diegoreymendez) [reply WordPressAuthenticator/Services/LoginFacade.swift:6]: Yeah, that shouldn't be there.

File: WordPressAuthenticator/Signin/LoginPrologueViewController.swift (hunk 3 above)
Comment (diegoreymendez) [reply WordPressAuthenticator/Signin/LoginPrologueViewController.swift]: Changed.

File: WordPressAuthenticator/Unified Auth/View Related/Password/PasswordViewController.swift (hunk 5 above)
Comment (diegoreymendez) [reply WordPressAuthenticator/Unified Auth/View Related/Password/PasswordViewController.swift]: Changed.

**Pull request merged**